### PR TITLE
[koa-shopify-auth] Set cookies with SameSite=none option

### PR DESCRIPTION
**Pull request by dimapaloskin**
## Description

Fixes (issue #1099)

In the latest versions of the chrome `koa-shopify-auth` users faced the problem of infinity redirection. Looks like it's because of [new changes](https://blog.chromium.org/2019/10/developers-get-ready-for-new.html) in `SameSite` behavior. Looks like an app can not access cookies from iframe if cookies were set up outside iframe. To fix that now we need to use `SameSite=None; Secure` settings for each cookie we want to receive inside an iframe. 

I propose to use `sameSite: 'none'` option in all `cookies.set` calls.

## Type of change

- [ ] `koa-shopify-auth` Patch: Bug/ Documentation fix (non-breaking change which fixes an issue or adds documentation)

## Checklist

- [ ] I have added a changelog entry, prefixed by the type of change noted above

### CODE REVIEW

File: packages/koa-shopify-auth/src/auth/create-enable-cookies-redirect.ts [hunk 1]
```diff
@@ -9,7 +9,9 @@ export default function createEnableCookiesRedirect(path: string) {
 
   return function topLevelOAuthRedirect(ctx: Context) {
     // This is to avoid a redirect loop if the app doesn't use verifyRequest or set the test cookie elsewhere.
-    ctx.cookies.set(TEST_COOKIE_NAME, '1');
+    ctx.cookies.set(TEST_COOKIE_NAME, '1', {
+      sameSite: 'none',
```
Comment (rmtngh) [inline packages/koa-shopify-auth/src/auth/create-enable-cookies-redirect.ts]: secure: true
https://www.chromestatus.com/feature/5088147346030592

> (Cookies marked SameSite=None should also be marked Secure.)

Comment (gil--) [reply packages/koa-shopify-auth/src/auth/create-enable-cookies-redirect.ts]: Also is this a safe change? Per https://github.com/Shopify/shopify_app/pull/851#issue-361762137 we may need to check the browser agent against a whitelist/blacklist.

> Additionally, SameSite=None (which we do not use today) is unfortunately interpreted incorrectly by specific browsers, so this PR handles such situations as well.

Comment (rmtngh) [reply packages/koa-shopify-auth/src/auth/create-enable-cookies-redirect.ts]: The ideal solution in my opinion would be adding a config option that receives cookie options and letting the devs add the logic around this themselves and remove it whenever other browsers (Safari) decided to fix it!

```
createShopifyAuth({
    apiKey: <SHOPIFY_API_KEY>,
    secret: <SECRET>,
    // If browser != "Safari"
    cookieOptions: {
        sameSite: 'none',
        secure: true
    }
  }),
```

**Comment by mkevinosullivan** [comment]
Thank you for contributing to the ongoing improvement of this library - we really appreciate it.

In order to optimize the support on this particular piece of functionality, we have moved it to its own repo.  As a result, I’m closing this PR because the underlying source has been moved to a dedicated new repo; `Shopify/koa-shopify-auth` (https://github.com/Shopify/koa-shopify-auth).

If you would like the PR to be considered for review, please re-create the change against the new `Shopify/koa-shopify-auth repo` and open a new PR there.

Thank you!

**Pull request closed without merging**